mautic/mautic
# DPMMA-2372 Email usages

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ N ]
| New feature/enhancement? (use the a.x branch)      | [ Y ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR adds information about email usage in segments, campaigns, forms, points and reports. Similar functionality already works for segments. 

This PR also brings UI improvements for this component - the table has been replaced with buttons so it's more compact and easier to click on the link.
![image](https://github.com/mautic/mautic/assets/8580942/7d3dc1a1-ee4b-4b48-97c6-d698e833d294)

The email usages are loaded asynchronously, so this should not affect the load time of the email view. 

A message has been added if the email/segment is not used elsewhere:
- For Emails: This email is not used by other components. 
- For Segments: Looks like this segment is not in use. 



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Add a new Email
3. Use the email in segment filter / campaign action / point trigger / point action / report
4. Check that the email usage is visible in the email view
